graph/dfs.cpp

```diff
@@ -1,26 +1,133 @@
+/**
```

Comment:
This file can be renamed to maybe `depth_first_search.cpp` or `depth_first_search_dfs.cpp`?

Reply:
good point, I'll rename it